sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -426,17 +427,22 @@ def setup(self):
 
   def process(self, batch, inference_args):
     start_time = _to_microseconds(self._clock.time_ns())
-    result_generator = self._model_handler.run_inference(
+    try:
+      result_generator = self._model_handler.run_inference(
         batch, self._model, inference_args)
-    predictions = list(result_generator)
+    except BaseException:
+      self._metrics_collector.failed_batches_counter.inc()
+      raise
+    else:
```

Comment:
We don't need to nest this clause in an else if we're raising the exception

Reply:
So when the exception is raised, that stops the method from executing further?

Reply:
Right - raising an exception (or in this case, reraising an exception) always causes function execution to terminate until you handle it (with an except clause). In this case, the exception will be reraised to the worker harness level before being retried (so all the code from the else statement on will be skipped).